### PR TITLE
Doc block fix for WP_Block_Patterns_Registry::register

### PR DESCRIPTION
**Pull request by kt-12**
Trac ticket: [https://core.trac.wordpress.org/ticket/59532](https://core.trac.wordpress.org/ticket/59532)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-block-patterns-registry.php [hunk 1]
```diff
@@ -79,6 +79,7 @@ final class WP_Block_Patterns_Registry {
 	 *                                   of the post types passed on the array. For all the other post types
 	 *                                   not part of the array the pattern is not available at all.
 	 *     @type string[] $templateTypes Optional. An array of template types where the pattern fits.
+	 *     @type string   $file_path     Optional. The path to the file containing the block pattern content.
```
Comment (joemcgill) [inline src/wp-includes/class-wp-block-patterns-registry.php]: Is this the full path, relative path, etc? Can we be more specific?

**Review by joemcgill — COMMENTED** [review]
Good update. The `$content` property should be updated to say that it is no longer required if there is a `$file_path` present. It would probably be good to move the new property up to just after $content as well, since they are so closely related. What do you think?

File: src/wp-includes/class-wp-block-patterns-registry.php (hunk 1 above)
Comment (jonnynews) [reply src/wp-includes/class-wp-block-patterns-registry.php]: Can we also update `*     @type string   $content       Required. Block HTML markup for the pattern.` so denote that is not required.

Comment (kt-12) [reply src/wp-includes/class-wp-block-patterns-registry.php]: @joemcgill @jonnynews  I have addressed both.

**Review by jonnynews — APPROVED** [review]

**Review by joemcgill — APPROVED** [review]

**Review by joemcgill — CHANGES_REQUESTED** [review]
After looking at this again, @kt-12 and I realized that the new property is the only one using snake case instead of camel case. Let's go ahead and fix those now for consistency and add, a `@since` annotation to the `register` method here.

```
 * @since 6.5.0 Added support for the `filePath` property.
```

**Review by joemcgill — APPROVED** [review]
🙏🏻 Thanks!

File: src/wp-includes/class-wp-block-patterns-registry.php [hunk 2]
```diff
@@ -79,6 +82,7 @@ final class WP_Block_Patterns_Registry {
 	 *                                   of the post types passed on the array. For all the other post types
 	 *                                   not part of the array the pattern is not available at all.
 	 *     @type string[] $templateTypes Optional. An array of template types where the pattern fits.
+	 *     @type string   $filePath     Optional. The full path to the file containing the block pattern content.
```
Comment (joemcgill) [inline src/wp-includes/class-wp-block-patterns-registry.php]:
```suggestion
	 *     @type string   $filePath      Optional. The full path to the file containing the block pattern content.
```

**Comment by joemcgill** [comment]
Merged in https://core.trac.wordpress.org/changeset/57731.

**Pull request closed without merging**